### DISCUSSION TITLE
[New3D] Support loading mesh from URDF using a relative path over http(s)

### DISCUSSION THREAD

**Discussion by jhurliman** [foxglove/community#453] [Q&A]
**Description**

Relative paths in URDFs to mesh resources fail to load when the URDF is fetched via http(s). Relative paths for other URL schemes such as package:// and file:// have not been tested.

- Version: `main`
- OS: all
- Data source (e.g. bag file, mcap file, rosbridge, ROS 1/2 native): none (add a custom URDF layer)
- ROS distro (if applicable): N/A

**Steps To Reproduce**

1. Create a URDF file that includes `<mesh filename="./mesh/robot.glb">`
2. Host the URDF at https://myassets.com/robot.urdf and the glTF at https://myassets.com/mesh/robot.glb
3. Create a custom URDF layer in `3D (Beta)` that points at https://myassets.com/robot.urdf
4. Observe the mesh fails to load, but when changing the filename to the absolute URL https://myassets.com/mesh/robot.glb it successfully loads

**Comment by defunctzombie** [comment]
Is this something we have ever supported? Or do we only support absolute paths for meshes (i.e. package:// https://) ?

**Comment by amacneil** [comment]
Changing to feature because I am not aware of any documentation saying we support this. We should still add it though.

**Comment by bertaveira** [comment]
I can confirm that relative paths using the `file://` scheme does not work either

**Comment by AltinayGrass** [comment]
Hello,

if we use `<mesh filename="file://$(find my_package)/meshes/mesh_file"/>` form. It cannot be shown on the 3D map

![image](https://github.com/user-attachments/assets/44398247-34c9-4d5c-9cf5-c653f78a006d)

**Reply by defunctzombie** [reply]
A filename with template arguments `$()`  is not supported. If you want to reference a package we support the `package://` scheme. See the docs here: https://docs.foxglove.dev/docs/visualization/panels/3d#resolution-of-urdf-assets-with-package-urls

**Reply by AltinayGrass** [reply]
Thanks for your reply. Thanks for your answer. I have another problem after loading meshes from https source. The robot is shown differently from the rviz view as below.
Any idea?
![image](https://github.com/user-attachments/assets/98e848c7-0072-420f-a5aa-f5404bd0fc6e)

**Reply by AltinayGrass** [reply]
Ok now. There is a option at view Mesh up-axis. I changed it to z-up.
